### PR TITLE
Refactor OreHelper to use "modern" APIs

### PR DESCRIPTION
**Pull request by Glease**
Close https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/9317

Please review with great care. I'm not sure if my code is functionally equivalent to the existing code.

### CODE REVIEW

File: src/main/java/appeng/util/item/OreHelper.java [hunk 1]
```diff
@@ -106,10 +87,12 @@ public OreReference isOre( final ItemStack itemStack )
 
 	boolean sameOre( final AEItemStack aeItemStack, final IAEItemStack is )
 	{
-		final OreReference a = aeItemStack.getDefinition().getIsOre();
```
Comment (repo-alt) [inline src/main/java/appeng/util/item/OreHelper.java:109]: Am I blind or was that always returning true?

**Review by repo-alt — APPROVED** [review]
In addition to speeding up startup this started to work correctly in some cases, which may break up existing setups. (Of course it sould be merged and tested live, if something breaks, then it was wrong already)

File: src/main/java/appeng/util/item/OreHelper.java (hunk 1 above)
Comment (Glease) [reply src/main/java/appeng/util/item/OreHelper.java:109]: This method don't see any real usage in ae2 code base, so 🤷‍♂️

**Comment by boubou19** [comment]
what kind of breaking should we expect to see?

File: src/main/java/appeng/util/item/OreHelper.java (hunk 1 above)
Comment (repo-alt) [reply src/main/java/appeng/util/item/OreHelper.java:109]: It is used by Platform.extractItemsByRecipe(). You probably mixed it with another one that is used only by AEItemStack.fuzzyComparison, which is in turn unused in AE2 itself (but is part of public API)

**Comment by repo-alt** [comment]
> what kind of breaking should we expect to see?

Platform.extractItemsByRecipe might theoretically be finding substitutions that are unwanted. e,g, if the pattern encodes <plankWood> and recipe allows also some special planks, not oredicted, previously they would be used, because oredict match was essentially always true, and recipe was always tested instead.

**Pull request merged**